src/regression.jl

```diff
@@ -1,35 +1,76 @@
-function conjugate(r::Regression, Y, Γ0::AbstractArray)
-    Γ0_ = Matrix(Γ0)
-    conjugate(r, Y, Mitosis.Gaussian{(:F,:Γ)}(similar(Γ0_, size(Γ0_, 1)), Γ0_))
+function conjugate(r::Union{Regression,Regression!}, Y, Γ0::AbstractArray)
+  Γ0_ = Matrix(Γ0)
+  conjugate(r, Y, Mitosis.Gaussian{(:F,:Γ)}(similar(Γ0_, size(Γ0_, 1)), Γ0_))
 end
 
 function conjugate(r::Regression, Y, prior::Gaussian)
-    @unpack k, fjac!, ϕ0func!, ϕ, ϕ0, y, y2, θ = r
-    @unpack g, pest = k
+  @unpack k, fjac, ϕ0func, θ = r
+  @unpack g, p = k
 
-    t = Y.t[1]
-    y .= Y.u[1]
+  t = Y.t[1]
+  y = Y.u[1]
 
-    calc_J!(ϕ, r, θ, t)
-    μ = prior.F
-    Γ = prior.Γ
+  ϕ = calc_J(y, r, θ, t)
+  μ = prior.F
+  Γ = prior.Γ
+
+  for i in 1:length(Y)-1
+    ϕ = calc_J(y, r, θ, t)
+    Gϕ = pinv(outer_(g(y, p, t)))*ϕ
+    zi = ϕ'*Gϕ
+    t2 = Y.t[i + 1]
+    y2 = Y.u[i + 1]
 
-    for i in 1:length(Y)-1
-        calc_J!(ϕ, r, θ, t)
-        Gϕ = pinv(outer_(g(y, pest, t)))*ϕ
-        zi = ϕ'*Gϕ
-        t2 = Y.t[i + 1]
-        y2 .= Y.u[i + 1]
+    ds = t2 - t
+
+    # compute intercept
+    if ϕ0func !== nothing
+      ϕ0 = ϕ0func(y, p, t)
+      μ = μ + Gϕ'*((y2 - y) .- ϕ0*ds)
+    else
+      μ = μ + Gϕ'*(y2 - y)
+    end
+
+    t = t2
+    y = y2
+    Γ = Γ + zi*ds
+
+  end
+  return Mitosis.Gaussian{(:F,:Γ)}(μ, Γ)
+end
 
-        ds = t2 - t
+function conjugate(r::Regression!, Y, prior::Gaussian)
```

Comment:
@mschauer  This conjugate function takes at the moment out-of-place functions `g` from `SDEProblem`  but does all the operations for the regression method inplace. Should this be changed to take only inplace functions `f!` or should this somehow support both?
Out of place functions  `g` could still be used with  `conjugate(r::Regression,..)` which however then also allocates.

Reply:
Can't immediately tell, can we make our minds up later or is there lock in?

Reply:
no big urgency at the moment. There is anyway a random test failure with the regression that I have to solve first before we can merge this PR.  I was just looking into the #10 and saw that SDEProblems with inplace functions are still missing and I wasn't sure which option one should prefer.